src/components/Pages/LoginPage/LoginPage.tsx

```diff
@@ -148,10 +160,22 @@ export const LoginPage = () => {
           const isClockSync = await checkClockSynchronization()
 
           if (isClockSync) {
-            await checkProfileAndRedirect(connectionData.account ?? '', referrer, () => {
-              redirect()
-              setShowConnectionModal(false)
-            })
+            const loginData = {
+              account: connectionData.account ?? '',
+              referrer,
+              redirect: () => {
+                redirect()
+                setShowConnectionModal(false)
+              }
+            }
+
+            if (isCheckingWebGPU) {
+              // Si WebGPU aún se está verificando, almacenar los datos para reintentar después
+              setPendingLoginData(loginData)
```

Comment:
What if instead of using the hook, we use the `checkWebGpuSupport` function here?